packages/angular-query-experimental/src/createBaseQuery.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  DestroyRef,
+  assertInInjectionContext,
+  computed,
+  effect,
+  inject,
+  signal,
+} from '@angular/core'
+import { notifyManager } from '@tanstack/query-core'
+import { QueryClientService } from './QueryClientService'
+import type { QueryClient, QueryKey, QueryObserver } from '@tanstack/query-core'
+import type { CreateBaseQueryOptions, CreateBaseQueryResult } from './types'
+
+/**
+ * Base implementation for `createQuery` and `createInfiniteQuery`.
+ * @internal
+ */
+export function createBaseQuery<
+  TQueryFnData,
+  TError,
+  TData,
+  TQueryData,
+  TQueryKey extends QueryKey,
+>(
+  options: () => CreateBaseQueryOptions<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryData,
+    TQueryKey
+  >,
+  Observer: typeof QueryObserver,
+  queryClient?: QueryClient,
+): CreateBaseQueryResult<TData, TError> {
+  assertInInjectionContext(createBaseQuery)
+  /** Load query client */
+  if (!queryClient) {
+    queryClient = inject(QueryClientService).useQueryClient()
+  }
+
+  /** Creates a signal that has the default options applied */
+  const defaultedOptionsSignal = computed(() => {
+    const defaultedOptions = queryClient!.defaultQueryOptions(options())
+    defaultedOptions._optimisticResults = 'optimistic'
+    return defaultedOptions
+  })
+
+  /** Creates the observer */
+  const observer = new Observer<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryData,
+    TQueryKey
+  >(queryClient, defaultedOptionsSignal())
```

Comment:
One issue with this approach is this will require the computed value to be evaluated immediatly. It may be possible that it may cause error.

example lets say the api call requires `id` passed to the component as input. This input is stored as signal and used to make the api call. If this computeds value is evaluated here, it would be undefined. 
Also similar issue will be there with signal component whenever we have that

I had implemented a weird approach to prevent this. I am initializing `Observer` with some empty option.

```
 const emptyQueryKey: QueryKey = [''] as const
const queryObserver = new Observer<any, any, any, any, QueryKey>(
    client,
    client.defaultQueryOptions({
      queryKey: emptyQueryKey,
      queryFn: () => {
        return Promise.resolve(false)
      },
      enabled: false,
    })
  )
```

Reply:
This is something I came across in React Query too. A solution is for the developer to set enabled to false until all information required by the query is present. If there is a better solution I would welcome it but I cannot think of anything.